src/ppx/grfx/dx12/dx12_swapchain.cpp

```diff
@@ -246,27 +243,23 @@ Result Swapchain::CreateApiObjects(const grfx::SwapchainCreateInfo* pCreateInfo)
 
     // Create color images
     {
-        auto ppxres = CreateColorImages(pCreateInfo->width, pCreateInfo->height, pCreateInfo->colorFormat, colorImages);
+        std::vector<void*> apiObjects;
+        for (const auto& colorImage : colorImages) {
+            apiObjects.push_back(colorImage);
+        }
+        auto ppxres = CreateColorImagesFromApiObjects(apiObjects);
         if (Failed(ppxres)) {
             return ppxres;
         }
     }
 
 #if defined(PPX_BUILD_XR)
-    {
-        for (size_t i = 0; i < depthImages.size(); ++i) {
-            grfx::ImageCreateInfo imageCreateInfo = grfx::ImageCreateInfo::DepthStencilTarget(pCreateInfo->width, pCreateInfo->height, pCreateInfo->depthFormat, grfx::SAMPLE_COUNT_1);
-            imageCreateInfo.pApiObject            = depthImages[i];
-
-            grfx::ImagePtr image;
-            Result         ppxres = GetDevice()->CreateImage(&imageCreateInfo, &image);
-            if (Failed(ppxres)) {
-                PPX_ASSERT_MSG(false, "image create failed");
-                return ppxres;
-            }
-
-            mDepthImages.push_back(image);
+    if (!depthImages.empty()) {
+        std::vector<void*> apiObjects;
+        for (const auto& depthImage : depthImages) {
+            apiObjects.push_back(depthImage);
         }
+        CreateDepthImagesFromApiObjects(apiObjects);
```

Comment:
nit: at least building for linux, seems like all those patterns can be shortened to either:
```cpp
std::vector<void*> apiObjects(depthImages.begin(), depthImages.end());
CreateDepthImagesFromApiObjects(apiObjects);
```
or
```cpp
CreateDepthImagesFromApiObjects({ depthImages.begin(), depthImages.end() });
```